gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiCategoryServiceImpl.java

```diff
@@ -130,6 +132,22 @@ private void removeCategory(ExecutionContext executionContext, Api api, String c
         }
     }
 
+    @Override
+    public void updateApiCategoriesToIds(io.gravitee.apim.core.api.model.Api api, String environmentId) {
+        if (api == null || api.getCategories() == null || api.getCategories().isEmpty()) {
+            return;
+        }
+        var categoryIds = api
+            .getCategories()
+            .stream()
+            .map(category -> categoryService.findById(category, environmentId))
```

Comment:
A user should be able to send us either the key or the id of a category. The call to the DB should be flexible with either.

Also, if we can do a bulk call instead of one for every category, it would be more efficient 👍

Reply:
@jourdiw regarding searching by keys and ids - we are covered here.
<img width="1067" alt="Screenshot 2024-05-20 at 10 58 06" src="https://github.com/gravitee-io/gravitee-api-management/assets/8080303/66370f76-61ec-4a17-bb3f-16120d1b97c9">

This methods first searches by id and then if not found - searches by keys.

Will add batching though as you suggested

Reply:
Resolved 1:1